### DISCUSSION TITLE
Сonnecting C code to the project.

### DISCUSSION THREAD

**Discussion by Artem-Fedulaev** [CosmosOS/Cosmos#2896] [Q&A] [answered]
**Hello everyone**, I'm developing my own OS and want to add C code to the project. I didn’t find any tutorials and decided to ask ChatGPT. She issued the code and it gives an error from IL2CPU. I will attach screenshots below.
![изображение](https://github.com/CosmosOS/Cosmos/assets/128837197/41cee823-78f3-46b5-bcdb-7e30169be012)
![изображение](https://github.com/CosmosOS/Cosmos/assets/128837197/85b064ad-4d4e-4331-94e5-6b14e810e402)
![изображение](https://github.com/CosmosOS/Cosmos/assets/128837197/31c21109-6441-48a4-b4ba-7a20c418931b)

**Comment by zarlo** [comment]
cosmos does not support this

**Reply by Artem-Fedulaev** [reply]
ChatGPT is unlikely to deceive.

**Reply by pleasenoban** [reply, marked as answer]
cosmos doesnt support dllimports and even if u try to load elfs into memory then jump into it, youll have to write your own stdlib impl
chatgpts dataset doesnt have too much info on cosmos just because its not very popular so its bound to hallucinate stuff

**Reply by retrievaaaa** [reply]
> ChatGPT is unlikely to deceive.

1) zarlo is a coredev. He knows Cosmos better than ChatGPT
2) Unlikely? Hallucinations are like the biggest problem LLMs have. Don't be stupid.

**Reply by Artem-Fedulaev** [reply]
Thanks everyone!

**Reply by Artem-Fedulaev** [reply]
I'm young and don't understand much about programming. I apologize if I was stupid somewhere

**Reply by zarlo** [reply]
like it could be done (on the linker step) but we do not support it and likely never will (but will one day have the ground works to let people do this in userland i dont know if we would have support it out of the box)
